lib/WebService/Async/Onfido/RateLimiter.pm

```diff
@@ -0,0 +1,152 @@
+package WebService::Async::Onfido::RateLimiter;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::Async::Onfido::RateLimiter - Module abstract
+
+=head1 SYNOPSIS
+
+    use WebService::Async::Onfido::RateLimiter;
+    my $instance = WebService::Async::Onfido::RateLimiter->new;
+
+=head1 DESCRIPTION
+
+=cut
+
+=head1 METHODS
+
+=cut
+
+use Carp qw();
+use Future;
+use mro;
+
+use parent qw(IO::Async::Notifier);
+
+=head1 NAME
+
+WebService::Async::CustomerIO::RateLimitter - This class provide possibility to limit amount
+of request in time interval
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+=cut
+
+sub _init {
+    my ($self, $args) = @_;
+    for my $k (qw(limit interval)) {
+        die "Missing required argument: $k" unless exists $args->{$k};
+        die "Invalid value for $k: $args->{$k}" unless int($args->{$k}) > 0;
+        $self->{$k} = delete $args->{$k} if exists $args->{$k};
```

Comment:
why are you checking twice? two lines above, there's an exception thrown if this condition isn't met